Real-Dev-Squad/website-status
# Fix mock service worker issue

## Problem:
Earlier, we were facing problems in working with msw because of the error mentioned in #411 

## Solution:
- Added an async call to initialise the mocks reference to the documentation used to fix this in next JS [here](https://github.com/vercel/next.js/tree/canary/examples/with-msw)
- (Additionally) Converted all the files required for mock service worker setup to `.ts` files

## How to use it for testing?:
1. Replace the current contents of .env.development to the contents below:
``` javascript
NEXT_PUBLIC_BASE_URL='https://localhost'
NEXT_PUBLIC_GITHUB_IMAGE_URL='https://raw.githubusercontent.com/Real-Dev-Squad/website-static/main/members'
NEXT_PUBLIC_API_MOCKING='ON'
```
2. Spin up the server and go to `http://localhost:3000`

Note that you may get Axios error on the first go. Try going to some other tab e.g., `idle users` and then, click on `tasks` again and also, the tabs that get response from an API with no event handler won't show data
Here's the video explaining the correct behavious as of now:
https://www.loom.com/share/12176c319bea4aeca25cd997de6c9bae
Ticket for a permanent fix of this https://github.com/Real-Dev-Squad/website-status/issues/444

## How to check if it is working right?
The data shown on the page should be the one that is mocked in handlers and not from the prod API
